packages/core/src/infer-proxy-admin.test.ts

```diff
@@ -0,0 +1,48 @@
+import test from 'ava';
+import { EthereumProvider } from './provider';
+import { inferProxyAdmin } from './infer-proxy-admin';
+
+const addr = '0x123';
+
+function makeProviderReturning(result: unknown): EthereumProvider {
+  return { send: (_method: string, _params: unknown[]) => Promise.resolve(result) } as EthereumProvider;
+}
+
+function makeProviderError(msg: string): EthereumProvider {
+  return {
+    send: (_method: string, _params: unknown[]) => {
+      throw new Error(msg);
+    },
+  } as EthereumProvider;
+}
+
+test('inferProxyAdmin returns true when owner looks like an address', async t => {
+  // abi encoding of address 0x1000000000000000000000000000000000000123
+  const provider = makeProviderReturning('0x0000000000000000000000001000000000000000000000000000000000000123');
+  t.true(await inferProxyAdmin(provider, addr));
+});
+
+test('inferProxyAdmin returns false when owner is more than 20 bytes', async t => {
```

Comment:
```suggestion
test('inferProxyAdmin returns false when owner is more than 32 bytes', async t => {
```

Reply:
This does return false if the result is more than 20 bytes though.  When [parsing to determine whether it is an address](https://github.com/OpenZeppelin/openzeppelin-upgrades/pull/1083/files#diff-3cea14ff2284ad223d8822435a111d3b7b7ed158001adfdc652a51df94ca1bd1R11), we check the left part of the hex string to ensure the higher order bytes (beyond 20 bytes) are 0.

Reply:
Clarified this in commit https://github.com/OpenZeppelin/openzeppelin-upgrades/pull/1083/commits/caf7553715191c75e481b34fc1f030d7ab966a69

Reply:
This test case is when some of the leftmost 12 bytes are non-zero, so is clearly not a padded address.  Updated as follows to make this clearer:
```js
test('inferProxyAdmin returns false when returned value is 32 bytes but clearly not an address', async t => {
  // some higher order bytes beyond 20 bytes are non-zero (the 'abc' in the below)
  const provider = makeProviderReturning('0x000000000000000000000abc1000000000000000000000000000000000000123');
  t.false(await inferProxyAdmin(provider, addr));
});
```